### ISSUE TITLE
probably offer broadcast failure: No offers collected for 440s

### ISSUE THREAD

**Issue by maaktweluit** [golemfactory/yagna-triage#85]
Artur - Linux+yapapi - No offers collected for 440s, stopped by ctrl+c (attach `blender-yapapi.log`)

```
    $ examples/blender/blender.py --driver zksync --subnet-tag devnet-beta.1 
    [2021-03-02 12:37:43,427 INFO yapapi] Using log file `blender-yapapi.log`; in case of errors look for additional information there
    yapapi version: 0.5.1-alpha.5
    Using subnet: devnet-beta.1, payment driver: zksync, and network: rinkeby

    [2021-03-02 12:38:04,569 WARNING yapapi.summary] No offers have been collected from the market for 20s. Make sure you're using the latest released versions of yagna and yapapi, and the correct subnet. Also make sure that the timeout for computing all tasks is within the 5 min to 30 min range.

    ...

    [2021-03-02 12:45:04,643 WARNING yapapi.summary] No offers have been collected from the market for 440s. Make sure you're using the latest released versions of yagna and yapapi, and the correct subnet. Also make sure that the timeout for computing all tasks is within the 5 min to 30 min range.
    ^CShutting down gracefully, please wait a short while or press Ctrl+C to exit immediately...
    [2021-03-02 12:45:05,206 INFO yapapi.executor] Waiting for all services to finish...
    [2021-03-02 12:45:05,231 WARNING yapapi.summary] Computation cancelled
    [2021-03-02 12:45:05,231 INFO yapapi.summary] Negotiated 0 agreements with 0 providers
    0 tasks computed, total time: 0:07:21.615492
    [2021-03-02 12:45:05,246 INFO yapapi.summary] Total cost: 0
    [2021-03-02 12:45:05,247 INFO yapapi.summary] Executor has shut down
    Shutdown completed, thank you for waiting!
```

[blender-yapapi-no-offers.log.gz](https://github.com/golemfactory/yagna-triage/files/6068624/blender-yapapi-no-offers.log.gz)

**Comment by maaktweluit** [comment]
Smells like heisenberg ( isolated user variant )

**Comment by azawlocki** [comment]
Yagna data dir containing one successful `blender.py` run (requestor logs from that run attached to https://github.com/golemfactory/yagna-triage/issues/84) and a few subsequent runs for which no offers have been gathered:

[yagna-data-dir.tar.gz](https://github.com/golemfactory/yagna-triage/files/6068843/yagna-data-dir.tar.gz)

**Comment by tworec** [comment]
Requestor node has received just two offers (via market protocol broadcast) between
`2021-03-02 11:28:00` and `2021-03-02 11:45:05`:
```
$ echo "select properties like '%devnet-beta.1%' is_devnet_beta_1, properties like '%community.4%' is_community_4, insertion_ts, creation_ts
from market_offer where insertion_ts between '2021-03-02 11:28:00' and '2021-03-02 11:45:05'" | sqlite3 -line market.db
is_devnet_beta_1 = 0
  is_community_4 = 1
    insertion_ts = 2021-03-02 11:34:32.327
     creation_ts = 2021-03-02 11:27:37.165297799

is_devnet_beta_1 = 0
  is_community_4 = 1
    insertion_ts = 2021-03-02 11:34:33.409
     creation_ts = 2021-03-02 11:27:38.033281347
```

but there was 70 offers subscribed during that period as shown by:
https://yastats.golem.network/explore?orgId=1&left=%5B%221614684480000%22,%221614685505000%22,%22Loki%22,%7B%22expr%22:%22%7Bjob%3D%5C%22devnet-beta.1%5C%22%7D%20%7C%3D%20%5C%22Subscribed%20new%20Offer%5C%22%22%7D%5D

![image](https://user-images.githubusercontent.com/12720209/109838431-722c6880-7c46-11eb-86c4-6a55898d71f1.png)

Offer broadcast is sent immediately after Offer subscription, and if it fails we should see log like this
```
log::warn!("Failed to bcast offer [{}]. Error: {}.", offer.id, e,);`
```

But it is not there
https://yastats.golem.network/explore?orgId=1&left=%5B%221614684480000%22,%221614685505000%22,%22Loki%22,%7B%22expr%22:%22%7Bjob%3D%5C%22devnet-beta.1%5C%22%7D%20%7C%3D%20%5C%22Failed%20to%20bcast%20offer%5C%22%22%7D%5D

![image](https://user-images.githubusercontent.com/12720209/109838731-b91a5e00-7c46-11eb-9612-127d3916962b.png)

And in the requestor's yagna log we can see two lines with `database locked` just before last offers broadcast reception
```
2021-03-02T11:33:28Z DEBUG ya_market::protocol::discovery] Propagating 1/1 unsubscribed Offers received from [0xeecbb30ab54d3bd896587cfe7ea0c40e54fa42ab].
[2021-03-02T11:34:02Z ERROR r2d2] database is locked
[2021-03-02T11:34:02Z ERROR r2d2] database is locked
[2021-03-02T11:34:32Z DEBUG ya_market::protocol::discovery] Propagating 1/1 Offers received from [0xe458bba25cf412c3563731919ff4121a689060dd].
[2021-03-02T11:34:32Z DEBUG ya_market::matcher::resolver] Resolving incoming Offer [571bc7b5e6664a8f80b8f7059ee496c2-9b8fd126469bccc9090018909cc30f63f62407202f423927c750e1e26d446d3b]
[2021-03-02T11:34:33Z DEBUG ya_market::protocol::discovery] Propagating 1/1 Offers received from [0xe458bba25cf412c3563731919ff4121a689060dd].
[2021-03-02T11:34:33Z DEBUG ya_market::matcher::resolver] Resolving incoming Offer [2bba33508d734519b2a38285cf4bbac2-5c657bc2cee87ba8e0af2623b7bff336731ff4443db0f736f97c63a0ea8913d8]
```

I'm not sure what is the cause, but it seems like offer broadcast is not working correctly here.
@nieznanysprawiciel @mfranciszkiewicz any thoughts?

**Comment by tworec** [comment]
closed via golemfactory/yagna#1120
